src/librustc_resolve/diagnostics.rs

```diff
@@ -93,6 +93,41 @@ mod bar {
 ```
 "##,
 
+E0254: r##"
+This error indicates that an imported symbol resolved to an already imported
+symbol.
+
+An example of this error:
+
+```
+// crate foo
+pub use foo::bar;
+pub mod foo; // note that there's a `pub` here
+```
+
+```
+// other program
+extern crate foo;
+
+use foo::*; // error, `foo::*` tries to bind `foo::foo` to `foo`
```

Comment:
I agree that it's more direct but it doesn't make clear that an import from an external crate can cause this problem too (and I say this because, reading your code, I wouldn't be able to figure out that the code I was writing were correct and the external lib I was using that exported something weird.)
Still, I will do some rewriting, thanks.